anms-core/anms/routes/ARIs/reports.py

```diff
@@ -102,42 +110,79 @@ async def report_ac(agent_id: str, correlator_nonce: int):
         exec_set_entry=["time"]
         if result:
             exec_set = result.entries.hex()
-            exec_set = await transcoder.transcoder_put_cbor_await("ari:0x"+exec_set)
-            exec_set =  exec_set['data']  
-            # format 
-            # TODO HANDLE RPTT and split up multiple entries 
-            "ari:/EXECSET/n=12345;(//ietf/dtnma-agent/CTRL/inspect(//ietf/dtnma-agent/EDD/sw-version))"
-            execset_pattern = r"ari:/EXECSET/n=.+;\((.*)\)"
-            match = re.match(execset_pattern,exec_set)
-            if match:
-                exec_set_entry.extend(match.group(1).split(';'))
-            else:
-                exec_set_entry.extend([exec_set])
-        final_res.append(exec_set_entry)
-
+            # use ACE to handle report set decoding  
+            in_text = '0x'+exec_set
+            try:
+                in_bytes = ace.cborutil.from_hexstr(in_text)
+                dec = ace.ari_cbor.Decoder()
+                ari = dec.decode(io.BytesIO(in_bytes))
+                ari = ace.nickname.Converter(ace.nickname.Mode.FROM_NN,  ace.AdmSet(cache_dir=False).db_session(), False)(ari)
```

Comment:
The nickname converter with an empty AdmSet will not translate anything, so this line can be left out.